pywbem/_subscription_manager.py

```diff
@@ -1284,16 +1309,87 @@ def _create_filter(self, server_id, source_namespace, query,
                         filter_inst = server.conn.GetInstance(filter_path)
                         self._owned_filters[server_id][i] = filter_inst
                     return filter_inst
-            filter_path = server.conn.CreateInstance(filter_inst)
+
+            filter_path = server.conn.CreateInstance(
+                filter_inst, namespace=server.interop_ns)
             filter_inst = server.conn.GetInstance(filter_path)
             self._owned_filters[server_id].append(filter_inst)
         else:
             # Responsibility to ensure it does not exist yet is with the user
-            filter_path = server.conn.CreateInstance(filter_inst)
+            filter_path = server.conn.CreateInstance(
+                filter_inst, namespace=server.interop_ns)
             filter_inst = server.conn.GetInstance(filter_path)
 
+        # If systemname does not yet exist, create it
+        if server_id not in self._systemnames:
+            self._systemnames[server_id] = filter_inst['SystemName']
+        else:
+            if self._systemnames[server_id] != filter_inst['SystemName']:
+                warnings.warn(
+                    _format("WBEM server may be returning different SystemName "
+                            "property for different operations: received  "
+                            "{0!A} and {0!A}. Owned filters may not be found.",
+                            self._systemnames[server_id],
+                            filter_inst['SystemName']),
+                    UserWarning)
+
         return filter_inst
 
+    def _get_systemname(self, server_id):
+        """
+        Get the SystemName returned for anything returned for this server_id.
+        This method gets the s
+        """
+        if server_id in self._systemnames:
+            return self._systemnames[server_id]
+
+        return None
```

Comment:
I think if the SystemName was not previously discovered, it should be obtained from the CIMOM instance.

Reply:
The dummy thing is a temp hack because we need subscription, etc. providers in the mock.

I like the idea of making the WBEMServer  SystemName and/or the entire CIMOM instance.  That eliminates the whole issue of this value without extra  processing cost.

Will do.

Reply:
In our call today, we decided that the WBEMServer class should be extended to have the entire CIMOM instance as a public property. May be in a separate PR, though.

Reply:
Modified to get SystemName from WBEMServer.cimom_inst object.  Also the 'Dummy' eliminated

Note that the the ObjectManager is already available as the property cimom_inst

DONE